### PR TITLE
txn: set last_change_ts only on 6.5+ versions

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Issue Number: Ref #13694

What's Changed:

```commit-message
Some old versions or components (TiKV < 5.0, TiFlash) cannot handle
unknown fields in Lock and Write. To avoid causing unexpected
problems, we add a feature gate to the new field.

We are not going to release this feature in TiKV 6.4, so I directly set
the minimal version to 6.5.
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by MyonKeminta** [comment]
Can this PR close https://github.com/pingcap/tiflash/issues/6238 ?

**Comment by CalvinNeo** [comment]
> Can this PR close [pingcap/tiflash#6238](https://github.com/pingcap/tiflash/issues/6238) ?

It can. However, we are also working to adapt this new lock type.

File: src/storage/mvcc/reader/scanner/forward.rs [hunk 1]
```diff
@@ -2366,6 +2366,16 @@ mod delta_entry_tests {
...
     fn test_mess() {
+        use pd_client::FeatureGate;
+
+        use crate::storage::txn::sched_pool::set_tls_feature_gate;
+
+        // Set version to 6.5.0 to enable last_change_ts.
+        // TODO: Remove this after TiKV version reaches 6.5
+        let feature_gate = FeatureGate::default();
+        feature_gate.set_version("6.5.0").unwrap();
+        set_tls_feature_gate(feature_gate);
```
[the remark this answers is not in the document]
Comment (sticnarf) [reply src/storage/mvcc/reader/scanner/forward.rs:2377]: It should not. The nextest framework run each test in a different process. The built-in Rust test framework runs each test in a different thread.

**Review by ekexium — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**